content/300-guides/050-database/900-turso.mdx

```diff
@@ -0,0 +1,163 @@
+
+## Connect and query your primary database
+
+1. Create an authentication token that will allow you to connect to the database:
+
+```sh
+turso db tokens create turso-prisma
+```
+
+1. Retrieve your database's connection string:
+
+```sh
+turso db show turso-prisma
+```
+
+1. Update your `.env` file with the authentication token and connection string:
+
+```text
+TURSO_AUTH_TOKEN="eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9..."
+TURSO_DATABASE_URL="libsql://turso-prisma-user.turso.io"
+```
+
+1. Enable the `driverAdapters` Preview feature flag:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = "file:./dev.db"
+}
+```
+
+1. Generate Prisma Client:
+
+```sh
+npx prisma generate
+```
+
+Install the libSQL database client and Prisma adapter for libSQL:
+
+```sh
+npm install @libsql/client @prisma/adapter-libsql
+```
+
+Update your Prisma Client instance with the following snippet:
+
+```ts
+import { PrismaClient } from '@prisma/client'
+import { PrismaLibSQL } from '@prisma/adapter-libsql'
+import { createClient } from '@libsql/client'
+
+const libsql = createClient({
+  url: `${process.env.TURSO_DATABASE_URL}`,
+  authToken: `${process.env.TURSO_AUTH_TOKEN}`,
+})
+
+const adapter = new PrismaLibSQL(libsql)
+const prisma = new PrismaClient({ adapter })
+```
+
+You can use Prisma Client as you normally would with full type-safety as you normally would in your project.
+
+<!-- TODO: how to connect to a remote database replica -->
+
+## How to connect and query from an embedded replica
+
+Turso supports [embedded replicas](https://blog.turso.tech/introducing-embedded-replicas-deploy-turso-anywhere-2085aa0dc242). Turso's embedded replicas enable you to have a copy of your primary, remote database _inside_ your application. This is similar to having an embedded or SQLite database.
+
+When your app initially establishes a connection to your database, the primary database will fulfill the query:
+
+![Embedded Replica: First remote read](./images/embedded-replica-remote-read.png)
+
+Turso will (1) create an embedded replica inside your application and (2) copy data from your primary database to the replica so it is locally available:
+
+![Embedded Replica: Remote DB Copy](./images/embedded-replica-create-replica.png)
+
+The embedded replica will fulfill subsequent read queries. The libSQL client provides a [`sync()`](<https://docs.turso.tech/libsql/client-access/javascript-typescript-sdk#client-capability-summary:~:text=an%20interactive%20transaction-,sync(),-Synchronize%20the%20embedded>) method which you can invoke to ensure the embedded replica's data remains fresh.
+
+![Embedded Replica: Local DB reads](./images/embedded-replica-read.png)
+
+With embedded replicas, this setup guarantees your app is fast because the data will be readily available locally.
```

Comment:
```suggestion
With embedded replicas, this setup guarantees a responsive application, because the data will be readily available locally and faster to access.
```